app/models/adviser.rb

```diff
@@ -29,6 +29,27 @@ class Adviser < ActiveRecord::Base
 
   validate :match_reference_number
 
+  after_save :geocode_if_needed
+
+  def full_street_address
+    [postcode, 'United Kingdom'].delete_if(&:blank?).join(', ')
+  end
+
+  def latitude=(value)
```

Comment:
These accessors and `geocode!` should be extracted and shared across `Firm` and `Adviser` as this stuff is identical. That's the best we can do for the moment, and then we'll revisit later. I'd quite like to punch together a macro-ish thing for this behaviour.